### PR TITLE
Revamping the helm Chart and implementing CI/CD

### PR DESCRIPTION
**Pull request by edmondop**
This PR implements CI/CD testing for the helm chart and addresses a couple of bugs discovered along the way:
- The tag for the image for different architecture override themselves because the metadata action ignored the flavor for specific types (see https://github.com/docker/metadata-action/issues/270)
- Implement docker build caching (see https://github.com/ArroyoSystems/arroyo/pull/271/files#diff-17e3400d876978d12bbad517dcf82312b54fa62723408b1ce011b2755458bdafR64 and https://github.com/ArroyoSystems/arroyo/pull/271/files#diff-17e3400d876978d12bbad517dcf82312b54fa62723408b1ce011b2755458bdafR95-R97 )
- Make the publishing of docker image generic and working with forks (see https://github.com/ArroyoSystems/arroyo/pull/271/files#diff-17e3400d876978d12bbad517dcf82312b54fa62723408b1ce011b2755458bdafR70)
- Fixes some styling issues on the Helm Chart that was failing lint (see https://github.com/ArroyoSystems/arroyo/pull/271/files#diff-00375794e129354bdafafd1f098385f1f65752bd9bbb5bb6c4410e2052746e3cR130-R141 )
- Fix an error in the helm chart  (see https://github.com/ArroyoSystems/arroyo/pull/271/files#diff-1e9d930a81e38d5d9b05f685528a044840126bab297513634024cf0fbe6bf85aR7-R8 )
- Remove one hardcoded value for the GRPC port (see https://github.com/ArroyoSystems/arroyo/pull/271/files#diff-1e9d930a81e38d5d9b05f685528a044840126bab297513634024cf0fbe6bf85aR7-R8 and https://github.com/ArroyoSystems/arroyo/pull/271/files#diff-5d82e8b7d1a16b8fd554980876f7373260a72ee4fe48f50d51a0a0ff339e4aa7R91-R99 )
- Add values to run the helm chart tests in CI/CD and one test to verify the GRPC server is up and running  and the test pipeline (see https://github.com/ArroyoSystems/arroyo/pull/271/files#diff-87b5fbf3f87fc3c6cdbdaf049ab34ed36df9cd69f471663e0a1cf40b48317390 , https://github.com/ArroyoSystems/arroyo/pull/271/files#diff-cc7da8cda14d8c9ad8156143814fdeb210cbc764a28a8499a4239d26471d7206R1-R14 , https://github.com/ArroyoSystems/arroyo/pull/271/files#diff-17e3400d876978d12bbad517dcf82312b54fa62723408b1ce011b2755458bdafR149-R213 )

Unfortunately my fork has no access to the secrets, so I think for this reason it fails. In my local fork everything works fine https://github.com/edmondop/arroyo/pull/2

### CODE REVIEW

File: k8s/arroyo/templates/tests/test-grpc-connection.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  name: "{{ include "arroyo.fullname" . }}-test-connection"
+  labels:
+    {{- include "arroyo.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": test
+spec:
+  containers:
+    - name: grpcurl
+      image: fullstorydev/grpcurl:v1.8.7-alpine
+      args: ['-plaintext', '{{ include "arroyo.fullname" . }}-api:{{ .Values.service.grpcPort }}','list']
```
Comment (mwylde) [inline k8s/arroyo/templates/tests/test-grpc-connection.yaml:13]: I think we'll need to change this to the HTTP endpoint now as the gRPC endpoint is gone in the API server; we might also want to test that the other services are up

**Review by mwylde — APPROVED** [review]
Thanks for the contribution. This looks great!

 There's one merge conflict right now, and I think the testing will need to be updated a bit now that the gRPC API is gone, once those are fixed I think we should be able to merge.

File: k8s/arroyo/templates/tests/test-grpc-connection.yaml (hunk 1 above)
Comment (edmondop) [reply k8s/arroyo/templates/tests/test-grpc-connection.yaml:13]: There was already a test for the HTTP endpoint, should this just be removed?

Comment (edmondop) [reply k8s/arroyo/templates/tests/test-grpc-connection.yaml]: This should address the problem with the services  [`554cab7` (#271)](https://github.com/ArroyoSystems/arroyo/pull/271/commits/554cab74c3ae4a7eece9254e4d92da1a302323c7) 
I will submit a separate commit for the output dir

File: k8s/arroyo/templates/tests/test-grpc-connection.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  name: "{{ include "arroyo.fullname" . }}-compiler-test-grpc-connection"
+  labels:
+    {{- include "arroyo.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": test
+spec:
+  containers:
+    - name: grpcurl
+      image: fullstorydev/grpcurl:v1.8.7-alpine
+      args: ['-plaintext', '{{ include "arroyo.fullname" . }}-api:{{ .Values.compiler.service.grpcPort }}','list']
```
Comment (mwylde) [inline k8s/arroyo/templates/tests/test-grpc-connection.yaml:13]: should this be `-compiler` (and below, `-controller`) instead of `-api`?

Comment (edmondop) [reply k8s/arroyo/templates/tests/test-grpc-connection.yaml]: Fixed with
[`af93b05` (#271)](https://github.com/ArroyoSystems/arroyo/pull/271/commits/af93b0544a2a347856846baeae08322681a26d2f)

**Review by mwylde — APPROVED** [review]
Looks great!

**Pull request merged**